core/email_api.php

```diff
@@ -734,14 +741,16 @@ function email_relationship_child_resolved_closed( $p_bug_id, $p_message_id ) {
 /**
  * Store email in queue for sending
  *
- * @param string $p_recipient Email recipient address.
- * @param string $p_subject   Subject of email message.
- * @param string $p_message   Body text of email message.
- * @param array  $p_headers   Array of additional headers to send with the email.
+ * @param string  $p_recipient Email recipient address.
+ * @param string  $p_subject   Subject of email message.
+ * @param string  $p_message   Body text of email message.
+ * @param array   $p_headers   Array of additional headers to send with the email.
+ * @param boolean $p_force     True to force sending of emails in shutdown function,
+ *                             even when using cronjob
  * @return integer|null
  */
-function email_store( $p_recipient, $p_subject, $p_message, array $p_headers = null ) {
-	global $g_email_stored;
+function email_store( $p_recipient, $p_subject, $p_message, array $p_headers = null, $p_force = false ) {
```

Comment:
I actually think providing a boolean flag is clearer and more meaningful from an API user's perspective.

I'm not sure about the flexibility argument, considering that EMAIL_SHUTDOWN_FORCE actually implies EMAIL_SHUTDOWN_GENERATED. 

To be technically correct, I probably should revise the assignment of *$g_email_shutdown_processing* in *email_store()* to always set the EMAIL_SHUTDOWN_GENERATED flag.